src/Api/HttpRestClient.php

```diff
@@ -0,0 +1,365 @@
+    {
+        /** @var array<string, array<string>> $result */
+        $result = $this->execute($this->consumerGet($consumer, '/subscription'), 'array<string, array<string>>');
+
+        return $result['topics'];
+    }
+
+    /** {@inheritDoc} */
+    public function consumerUnsubscribe(Consumer $consumer) : void
+    {
+        $this->execute($this->requestFactory->createRequest('DELETE', $consumer->baseUri . '/subscription'));
+    }
+
+    /** {@inheritDoc} */
+    public function consumerAssignPartitions(Consumer $consumer, iterable $partitions) : void
+    {
+        $this->execute($this->consumerPost($consumer, '/assignments', $this->serialize(['partitions' => $partitions])));
+    }
+
+    /** {@inheritDoc} */
+    public function getConsumerAssignedPartitions(Consumer $consumer) : array
+    {
+        /** @var array<string, array<AssignedPartition>> $result */
+        $result = $this->execute(
+            $this->consumerGet($consumer, '/assignments'),
+            sprintf('array<string, array<%s>>', AssignedPartition::class)
+        );
+
+        return $result['partitions'];
+    }
+
+    /** {@inheritDoc} */
+    public function consumerSeek(Consumer $consumer, iterable $offsets) : void
+    {
+        $this->execute($this->consumerPost($consumer, '/positions', $this->serialize(['offsets' => $offsets])));
+    }
+
+    /** {@inheritDoc} */
+    public function consumerSeekStart(Consumer $consumer, iterable $partitions) : void
+    {
+        $this->execute(
+            $this->consumerPost($consumer, '/positions/beginning', $this->serialize(['partitions' => $partitions]))
+        );
+    }
+
+    /** {@inheritDoc} */
+    public function consumerSeekEnd(Consumer $consumer, iterable $partitions) : void
+    {
+        $this->execute(
+            $this->consumerPost($consumer, '/positions/end', $this->serialize(['partitions' => $partitions]))
+        );
+    }
+
+    /** {@inheritDoc} */
+    public function getConsumerMessages(Consumer $consumer, ?int $timeout = null, ?int $maxBytes = null) : array
+    {
+        $parameters = [];
+        if ($timeout !== null) {
+            $parameters['timeout'] = $timeout;
+        }
+
+        if ($maxBytes !== null) {
+            $parameters['maxBytes'] = $maxBytes;
+        }
+
+        $request = $this->consumerGet($consumer, '/records');
+        $request = $request->withUri($request->getUri()->withQuery(http_build_query($parameters)), true);
+
+        /** @var array<Message> $messages */
+        $messages = $this->execute($request, sprintf('array<%s>', Message::class));
+
+        return $messages;
+    }
+
+    /** {@inheritDoc} */
+    public function getBrokers() : array
+    {
+        /** @var array<string, array<int>> $result */
+        $result = $this->execute($this->get('/brokers'), 'array<string, array<int>>');
+
+        return $result['brokers'];
+    }
+
+    private function get(string $path, ?UriInterface $baseUri = null) : RequestInterface
```

Comment:
I would move get() and port() above execute()

Reply:
it is above execute ... the other two methods between `get`/`post` and `execute` are the other two methods which do the same thing, but for consumer endpoints